### PR TITLE
Convert dialogs policy article.

### PR DESCRIPTION
**Pull request by jpmedley**

### CODE REVIEW

File: src/content/en/updates/2017/03/dialogs-policy.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+project_path: /web/_project.yaml
+book_path: /web/updates/_book.yaml
+description: Synchronous, app-modal JavaScript dialogs are commonly (and unfortunately) sued to harm users. Because of this, the Chromium team highly recommends that you not use JavaScript dialogs.
```
Comment (avidrissman) [inline src/content/en/updates/2017/03/dialogs-policy.md]: "sued to harm users" -> "used to harm users"

File: src/content/en/updates/2017/03/dialogs-policy.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+{# wf_updated_on: 2017-03-24 #}
+{# wf_published_on: 2017-03-24 #}
+{# wf_tags: policy,dialog,javascript #}
+{# wf_featured_image: /web/updates/images/generic/warning.png #}
+{# wf_featured_snippet: Synchronous, app-modal JavaScript dialogs are commonly (and unfortunately) sued to harm users. Because of this, the Chromium team highly recommends that you not use JavaScript dialogs. #}
```
Comment (avidrissman) [inline src/content/en/updates/2017/03/dialogs-policy.md]: ditto

File: src/content/en/updates/2017/03/dialogs-policy.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+should be used. For obtaining user input, the
+[HTML &lt;dialog&gt; element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog)
+should be used. For XSS proofs-of-concept, devtool’s
+`console.log(document.origin)` can be used.
+
+As for `onbeforeunload`, it should be noted that it is _already_ unreliable. As
+Ilya Grigorik [points out](https://www.igvita.com/2015/11/20/dont-lose-user-and-app-state-use-page-visibility/),
+“You _cannot rely_ on `pagehide`, `beforeunload`, and `unload` events to fire on
+“mobile platforms.” If you need to save state, you should use the [Page
+“Visibility API](http://w3c.github.io/page-visibility/#introduction).
```
Comment (avidrissman) [inline src/content/en/updates/2017/03/dialogs-policy.md]: The open quotes on lines 46 and 47 show up in the output. You need to clean the formatting here and verify it comes out correctly.

File: src/content/en/updates/2017/03/dialogs-policy.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+(It was also removed by Safari starting with Safari 9.1 and in Firefox 4.)
+
+It is planned that `alert()/confirm()/prompt()` dialogs will not be app-modal,
+but rather [will be dismissed when their tab is switched from](http://crbug.com/629964).
+(Safari 9.1 already does this.)
+
+We are investigating the possibility of
+[gating `alert()/confirm()/prompt()/onbeforeunload` dialogs on site engagement](https://groups.google.com/a/chromium.org/d/msg/blink-dev/waPXPhKqbJ0/wYVJlFKkBgAJ),
+so if the user isn’t engaging with the page, the page’s dialogs will not be
+shown. Details have yet to be ironed out.
```
Comment (avidrissman) [inline src/content/en/updates/2017/03/dialogs-policy.md]: This doc was written a while ago. Right now the plan is to gate those dialogs on a page having a user gesture. Will this doc be able to be updated when this goes into action?

File: src/content/en/updates/2017/03/dialogs-policy.md (hunk 1 above)
Comment (petele) [inline src/content/en/updates/2017/03/dialogs-policy.md]: 👍

File: src/content/en/updates/2017/03/dialogs-policy.md (hunk 2 above)
Comment (petele) [inline src/content/en/updates/2017/03/dialogs-policy.md]: 👍

File: src/content/en/updates/2017/03/dialogs-policy.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+Because of these changes, if your site uses dialogs, it is highly recommended
+that you move to using the earlier-mentioned alternatives so that this will not
+affect you.
+
+Thank you for your understanding. We appreciate your help in moving the web
```
Comment (petele) [inline src/content/en/updates/2017/03/dialogs-policy.md]: Nit, I'd remove this.

File: src/content/en/updates/2017/03/dialogs-policy.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+“Visibility API](http://w3c.github.io/page-visibility/#introduction).
+
+## Changes
+
+The ability for a page to specify the `onbeforeunload` string was
+[removed in Chrome 51](https://www.chromestatus.com/feature/5349061406228480).
+(It was also removed by Safari starting with Safari 9.1 and in Firefox 4.)
+
+It is planned that `alert()/confirm()/prompt()` dialogs will not be app-modal,
+but rather [will be dismissed when their tab is switched from](http://crbug.com/629964).
```
Comment (petele) [inline src/content/en/updates/2017/03/dialogs-policy.md]: Nit, please try to use HTTPS urls where possible

File: src/content/en/updates/2017/03/dialogs-policy.md (hunk 4 above)
Comment (jpmedley) [inline src/content/en/updates/2017/03/dialogs-policy.md]: Yes.

Comment (jpmedley) [inline src/content/en/updates/2017/03/dialogs-policy.md]: You can ping me and I'll update it, or you can submit your own PR. (I apologize again for the slow turnaround on publishing this. Circumstances on our end were unusual.)

File: src/content/en/updates/2017/03/dialogs-policy.md (hunk 5 above)
Comment (jpmedley) [inline src/content/en/updates/2017/03/dialogs-policy.md]: @avidrissman FYI.

**Comment by jpmedley** [comment]
@petele @avidrissman PTAL

File: src/content/en/updates/2017/03/dialogs-policy.md (hunk 5 above)
Comment (avidrissman) [inline src/content/en/updates/2017/03/dialogs-policy.md]: No objection.

**Pull request merged**